### DISCUSSION TITLE
creating velero image with USER numeric.

### DISCUSSION THREAD

**Discussion by vijaynetha** [vmware-tanzu/velero#4189] [Community support Q&A]
Hi team,

i have cloned velero code from this link https://github.com/vmware-tanzu/velero/tree/main
iam trying to create new velero image with user 1000:1000. so i changed USER nonroot:nonroot to USER 1000:1000 in Dockerfile.

while building image i am getting some issue "./hack/download-restic.sh: line 26: BIN: unbound variable".

build is failing "./hack/download-restic.sh" because of this shell script.

please help  me on this issue i need velero image with USER numeric.

docker build -t velero:v1.6.3 .
Sending build context to Docker daemon    9.9MB
Step 1/24 : FROM golang:1.16 as builder-env
 ---> 828b0508ae64
Step 2/24 : ARG GOPROXY
 ---> Using cache
 ---> e57407fb272c
Step 3/24 : ARG PKG
 ---> Using cache
 ---> b21ed2db1816
Step 4/24 : ARG VERSION
 ---> Using cache
 ---> 1651be0f341b
Step 5/24 : ARG GIT_SHA
 ---> Using cache
 ---> 9f9c4b906f51
Step 6/24 : ARG GIT_TREE_STATE
 ---> Using cache
 ---> 2f39e0ae539e
Step 7/24 : ARG REGISTRY
 ---> Using cache
 ---> 5ce90cf81efc
Step 8/24 : ENV CGO_ENABLED=0     GO111MODULE=on     GOPROXY=${GOPROXY}     LDFLAGS="-X ${PKG}/pkg/buildinfo.Version=${VERSION} -X ${PKG}/pkg/buildinfo.GitSHA=${GIT_SHA} -X ${PKG}/pkg/buildinfo.GitTreeState=${GIT_TREE_STATE} -X ${PKG}/pkg/buildinfo.ImageRegistry=${REGISTRY}"
 ---> Using cache
 ---> c49073969aa8
Step 9/24 : WORKDIR /go/src/github.com/vmware-tanzu/velero
 ---> Using cache
 ---> 786c5f29f4b9
Step 10/24 : COPY . /go/src/github.com/vmware-tanzu/velero
 ---> Using cache
 ---> dfde86455aee
Step 11/24 : RUN apt-get update && apt-get install -y bzip2
 ---> Using cache
 ---> ac5649dc8d54
Step 12/24 : FROM builder-env as builder
 ---> ac5649dc8d54
Step 13/24 : ARG TARGETOS
 ---> Using cache
 ---> fccb0753e45a
Step 14/24 : ARG TARGETARCH
 ---> Using cache
 ---> b91c72fc6df5
Step 15/24 : ARG TARGETVARIANT
 ---> Using cache
 ---> f43fd0d0e781
Step 16/24 : ARG PKG
 ---> Using cache
 ---> a39f07e35c2d
Step 17/24 : ARG BIN
 ---> Using cache
 ---> ec6e661dcc75
Step 18/24 : ARG RESTIC_VERSION
 ---> Using cache
 ---> e6f8d4efaeed
Step 19/24 : ENV GOOS=${TARGETOS}     GOARCH=${TARGETARCH}     GOARM=${TARGETVARIANT}
 ---> Using cache
 ---> fd1fafdb7e45
Step 20/24 : RUN mkdir -p /output/usr/bin &&     bash ./hack/download-restic.sh &&     export GOARM=$( echo "${GOARM}" | cut -c2-) &&     go build -o /output/${BIN}     -ldflags "${LDFLAGS}" ${PKG}/cmd/${BIN}
 ---> Running in e4392ab14ad5
./hack/download-restic.sh: line 26: BIN: unbound variable
The command '/bin/sh -c mkdir -p /output/usr/bin &&     bash ./hack/download-restic.sh &&     export GOARM=$( echo "${GOARM}" | cut -c2-) &&     go build -o /output/${BIN}     -ldflags "${LDFLAGS}" ${PKG}/cmd/${BIN}' returned a non-zero code: 1

**Comment by zubron** [comment]
Please see the Makefile target `all-containers`: https://github.com/vmware-tanzu/velero/blob/main/Makefile#L123-L125
This will invoke Docker will with the correct parameters and build both images.
